### PR TITLE
feat: Retrieve all catalog data, not just programs

### PR DESCRIPTION
**Pull request by tuchfarber**
In order to maintain better data integrity and specifically to avoid
deleting course runs attached to grades, this syncs *all* catalog data
to Credentials, not just the data that exists in the /programs/ endpoint

### CODE REVIEW

File: credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Catalog Data Sync
+=================
+
+Status
+------
+Accepted
+
+Background
+----------
+The catalog app previously only synced programs-related data in addtion to pathways. In code, programs-related data was defined as "any data that is returned in the programs resource of the Discovery API". If the data was copied once, then excluded from the API, the data would still exist in Credentials because we didn't delete data.
```
Comment (crice100) [inline credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst:10]: Nit: "addition"

File: credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Decision
+--------
+The catalog app will now sync all organizations, courses, course runs, programs, and pathways from the Discovery service. Since adding the option to delete outdated information in the copy_catalog script, this full copy is required to make sure a course run isn't removed from a program that has existing grade data linked to it. In addition, all Discovery data will be needed as we transition course certs from LMS to Credentials.
+
+In order to sufficently link all Discovery data, the data will need to be copied in a specific order:
```
Comment (crice100) [inline credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst:16]: Nit: "sufficiently"

File: credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+------
+Accepted
+
+Background
+----------
+The catalog app previously only synced programs-related data in addtion to pathways. In code, programs-related data was defined as "any data that is returned in the programs resource of the Discovery API". If the data was copied once, then excluded from the API, the data would still exist in Credentials because we didn't delete data.
+
+Decision
+--------
+The catalog app will now sync all organizations, courses, course runs, programs, and pathways from the Discovery service. Since adding the option to delete outdated information in the copy_catalog script, this full copy is required to make sure a course run isn't removed from a program that has existing grade data linked to it. In addition, all Discovery data will be needed as we transition course certs from LMS to Credentials.
```
Comment (crice100) [inline credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst:14]: Suggestion: maybe soften this last bit to imply that we may transition course certificates to credentials in the future

File: credentials/apps/catalog/utils.py [hunk 4]
```diff
@@ -36,188 +59,282 @@ def __init__(self, site, api_client, catalog_api_url, page_size):
...
+        Returns:
+            str: A log of the changes to the data. Useful when the caller needs to print (not log) the data to the
+                console
+        """
         logger.info(f"Copying catalog data for site {self.site.domain}")
-        self._fetch_programs()
-        logger.info("Finished copying programs.")
-        self._fetch_pathways()
+        # fetch organizations
+        self.fetch_resource("organizations", self._parse_organization)
```
Comment (crice100) [inline credentials/apps/catalog/utils.py:96]: I don't suppose we have constants for these anywhere (instead of string values)?

**Review by crice100 — APPROVED** [review]
Love that you wrote doc along with this change!

File: credentials/apps/catalog/utils.py (hunk 4 above)
Comment (tuchfarber) [reply credentials/apps/catalog/utils.py]: Switched to constants :)

File: credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst (hunk 3 above)
Comment (tuchfarber) [reply credentials/apps/catalog/docs/decisions/0001-sync-all-catalog-data.rst]: Fixed!

**Pull request merged**